src/Domain/Infra/Doctrine/DomainEntityRepositoryTrait.php

```diff
@@ -144,4 +176,33 @@ private function addFieldCriteria(QueryBuilder $qb, array $fields, bool $or = fa
 
         $qb->andWhere($where);
     }
+
+    private function isValidEntityId(array $ids): bool
+    {
+        if (count($ids) !== $this->numIdFields) {
+            return false;
+        }
+
+        $metadataFactory = $this->em->getMetadataFactory();
+
+        return count(array_filter($ids, function ($id) use ($metadataFactory): bool {
+            if ($id instanceof DomainIdInterface) {
+                return !$id->isEmpty();
+            }
+
+            if (is_object($id)) {
+                $class = get_class($id);
+
+                foreach (($metadata = $metadataFactory->getMetadataFor($this->class))->getIdentifierFieldNames() as $idFieldName) {
+                    if ($class === $metadata->getAssociationTargetClass($idFieldName)) {
+                        return true;
+                    }
+                }
+
+                return method_exists($id, '__toString');
+            }
+
+            return is_scalar($id);
+        })) === $this->numIdFields;
```

Comment:
Little nitpicking, what if the filter is more than numFields? It would still iterate over the rest of the array. What if the amount of ids are so low, that even with the remaining few items, it won't ever reach the total?

Reply:
hm not sure i follow, the method is here to validate a (composite) primary id for the current repository class (it's only used in `find() + exists()`). We expect the no. of  id's to be equal to the no. of defined/available id fields (numIdFields). 

Filtering out empty ids avoids an unintended `WHERE pk_id = null` (in case the mapping allows such a definition), as `find() + exists()` are design to be used with PK values _only_ (which are non-empty).

Reply:
Updated.